it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/pubsub/PubsubResourceManager.java

```diff
@@ -394,6 +400,50 @@ private static <T> RetryPolicy<T> retryOnDeadlineExceeded() {
         .build();
   }
 
+  /**
+   * Collects the performance metrics for the pubsub resources.
+   *
+   * @param monitoringClient Monitoring client
+   * @param metrics The pubsub metrics will be populated in this map
+   */
+  public void getMetrics(
+      @NonNull MonitoringClient monitoringClient, @NonNull Map<String, Double> metrics) {
```

Comment:
Why should the consumer of a `Pubsub` resource care or know about the `monitoringClient`? 

As a user, I am using a library to manage by pubsub resources and shouldn't have to care about how it does the things that are internal to it. 

It should be initialized internally (you can do a lazy initialization if you want).

Reply:
Changed the implementation to accept the monitoring client when initializing the PubsubResourceManager. 

The idea is, the Load test (or integation test) would have an ensemble of RMs which are required to execute the test. If a resourceManager like Pubsub requires another RM like MonitoringClient to perform some functions, then the test can pass in the MonitoringClient to Pubsub RM during its creation.